packages/ua-devtools/src/oapp/config.ts

```diff
@@ -125,24 +125,32 @@ export const configureSendLibraries: OAppConfigurator = withOAppLogger(
     }
 )
 
-export const configureReceiveLibraries: OAppConfigurator = async (graph, createSdk) =>
-    flattenTransactions(
-        await Promise.all(
-            graph.connections.map(async ({ vector: { from, to }, config }): Promise<OmniTransaction[]> => {
+export const configureReceiveLibraries: OAppConfigurator = withOAppLogger(
+    createConfigureEdges(
+        withOAppLogger(
+            async ({ vector: { from, to }, config }, sdk): Promise<OmniTransaction[]> => {
+                const logger = createOAppLogger()
+
                 if (config?.receiveLibraryConfig == null) {
+                    logger.verbose(`receiveLibraryConfig not set for ${formatOmniVector({ from, to })}, skipping`)
                     return []
                 }
 
-                const oappSdk = await createSdk(from)
-                const endpointSdk = await oappSdk.getEndpointSDK()
+                const endpointSdk = await sdk.getEndpointSDK()
                 const [currentReceiveLibrary, isDefaultLibrary] = await endpointSdk.getReceiveLibrary(
                     from.address,
                     to.eid
                 )
 
                 if (!isDefaultLibrary && currentReceiveLibrary === config.receiveLibraryConfig.receiveLibrary) {
+                    logger.verbose(
+                        `Current recieveLibrary is not default and is already set to receiveLibraryConfig.receiveLibrary for ${formatOmniVector({ from, to })}, skipping`
+                    )
                     return []
                 }
+
+                logger.verbose(`Setting recieveLibrary for ${formatOmniVector({ from, to })}`)
```

Comment:
will also specify which `receiveLibrary` is being set here with `${config.receiveLibraryConfig.receiveLibrary}`